src/utils/mod.rs

```diff
@@ -190,11 +202,22 @@ pub fn new_cmark_parser(text: &str, curly_quotes: bool) -> Parser<'_, '_> {
 }
 
 pub fn render_markdown_with_path(text: &str, curly_quotes: bool, path: Option<&Path>) -> String {
+    render_markdown_with_abs_path(text, curly_quotes, path, None)
+}
+
+pub fn render_markdown_with_abs_path(
+    text: &str,
+    curly_quotes: bool,
+    path: Option<&Path>,
+    abs_url: Option<&String>,
```

Comment:
The `&String` type should normally not be used — `&str` is more flexible for the callers since they can pass in both string literals (that would simplify the tests below) and borrowed owned strings.